File: app/src/main/java/com/dhsdevelopments/potato/clientapi/channel2/Channel.java
package com.dhsdevelopments.potato.clientapi.channel2;

import com.google.gson.annotations.SerializedName;

public class Channel
{
    @SerializedName( "id" )
    public String id;

    @SerializedName( "name" )
    public String name;

    @SerializedName( "hide" )
    public boolean hide;

    @SerializedName( "group" )
    public String groupId;

    @SerializedName( "group_type" )
    public String groupType;

    @SerializedName( "unread_count" )
    public int unreadCount;

    @Override
    public String toString() {
        return "Channel[" +
                       "id='" + id + '\'' +
                       ", name='" + name + '\'' +
                       ", hide=" + hide +
                       ", groupId='" + groupId + '\'' +
                       ", groupType='" + groupType + '\'' +
                       ", unreadCount=" + unreadCount +
                       ']';
    }
}

File: app/src/main/java/com/dhsdevelopments/potato/clientapi/channel2/Channel.kt
package com.dhsdevelopments.potato.clientapi.channel2

import com.google.gson.annotations.SerializedName

class Channel {
    @SerializedName("id")
    lateinit var id: String

    @SerializedName("name")
    lateinit var name: String

    @SerializedName("hide")
    var hide: Boolean = false

    @SerializedName("group")
    lateinit var groupId: String

    @SerializedName("group_type")
    lateinit var groupType: String

    @SerializedName("unread_count")
    var unreadCount: Int = 0

    @SerializedName("private_user")
    var privateUser: String? = null

    override fun toString(): String {
        return "Channel[id='$id', name='$name', hide=$hide, groupId='$groupId', groupType='$groupType', unreadCount=$unreadCount, privateUser='$privateUser']"
    }
}
